src/lib.rs

```diff
@@ -126,11 +127,11 @@ impl Importer for CheckImporter {
         let TypecheckValue { expr, typ } =
             try!(macro_value.typecheck(compiler, vm, module_name, input));
 
-        let lines = source::Lines::new(input);
+        let lines = source::Lines::new(input.as_bytes().iter().cloned());
         let (metadata, _) = gluon::check::metadata::metadata(&*vm.global_env().get_env(), &expr);
         self.0.lock().unwrap().insert(
             module_name.into(),
-            Module {
+            self::Module {
```

Comment:
There is now a struct named `Module`  in `gluon::compiler_pipeline` that clashes with the `Module` struct defined in this module. I don't know if this is the preferred way to resolve the clash, or if we should change `use gluon::compiler_pipeline::*` to specify only what we want to import.

Reply:
Nah, that is fine. I will probably end up refactoring/breaking up `src/lib.rs` sometime soon anyway so there is no need for it to be perfect.